crates/relayer-cli/src/commands/evidence.rs

```diff
@@ -0,0 +1,795 @@
+    if !counterparty_client_is_frozen {
+        info!(
+            "will submit light client attack evidence to client `{}` on counterparty chain `{}`",
+            counterparty_client_id,
+            counterparty.id(),
+        );
+
+        let msg = MsgSubmitMisbehaviour {
+            client_id: counterparty_client_id.clone(),
+            misbehaviour: misbehaviour.to_any(),
+            signer,
+        }
+        .to_any();
+
+        msgs.push(msg);
+    }
+
+    if msgs.is_empty() {
+        warn!(
+            "skipping light client attack evidence for client `{}` on counterparty chain `{}`",
+            counterparty_client_id,
+            counterparty.id()
+        );
+
+        warn!("reason: no messages to submit");
+
+        return Ok(());
+    }
+
+    let tracked_msgs = TrackedMsgs::new_static(msgs, "light_client_attack_evidence");
+    let responses = counterparty.send_messages_and_wait_check_tx(tracked_msgs)?;
+
+    match responses.first() {
+        Some(response) if response.code.is_ok() => {
+            info!(
+                "successfully submitted light client attack evidence for client `{}` to counterparty chain `{}`, tx hash: {}",
+                counterparty_client_id,
+                counterparty.id(),
+                response.hash
+            );
+
+            Ok(())
+        }
+        Some(response) => Err(eyre::eyre!(
+            "failed to submit light client attack evidence to counterparty chain `{}`: {response:?}",
+            counterparty.id()
+        )),
+
+        None => Err(eyre::eyre!(
+            "failed to submit light client attack evidence to counterparty chain `{}`: no response from chain",
+            counterparty.id()
+        )),
+    }
+}
+
+fn has_consensus_state(
+    chain: &BaseChainHandle,
+    client_id: &ClientId,
+    consensus_height: Height,
+) -> bool {
+    use ibc_relayer::chain::requests::QueryConsensusStateRequest;
+
+    let res = chain.query_consensus_state(
+        QueryConsensusStateRequest {
+            client_id: client_id.clone(),
+            consensus_height,
+            query_height: QueryHeight::Latest,
+        },
+        IncludeProof::No,
+    );
+
+    res.is_ok()
+}
+
+fn is_counterparty_provider(
+    chain: &CosmosSdkChain,
+    counterparty_chain_handle: &BaseChainHandle,
+) -> bool {
+    if chain.config().ccv_consumer_chain {
+        let consumer_chains = counterparty_chain_handle
+            .query_consumer_chains()
+            .unwrap_or_default(); // If the query fails, use an empty list of consumers
+
+        // FIXME: Do do we need to check if the client id also matches?
+        // ie. is it okay to submit a `MsgSubmitIcsConsumerMisbehaviour` to all clients
+        // of the provider chain, or should we only do this for the CCV client, and
+        // use the standard message for other clients?
+        consumer_chains
+            .iter()
+            .any(|(chain_id, _)| chain_id == chain.id())
```

Comment:
@ancazamfir Do do we need to check if the client id also matches? ie. is it okay to submit a `MsgSubmitIcsConsumerMisbehaviour` to all clients of the provider chain, or should we only do this for the CCV client, and use the standard message for other clients?

Probably the latter, right?

Reply:
Yes, I think so.  @sainoe could you confirm?